src/ert/_c_wrappers/enkf/ert_run_context.py

```diff
@@ -12,42 +14,65 @@
 @dataclass
 class RunContext:
     sim_fs: "EnkfFs"
-    target_fs: "EnkfFs" = None
-    mask: List[bool] = field(default_factory=list)
-    paths: List[str] = field(default_factory=list)
-    jobnames: List[str] = field(default_factory=list)
+    path_format: str
+    format_string: str
+    runpath_file: Path
+    initial_mask: List[bool] = field(default_factory=list)
+    global_substitutions: Optional[Dict[str, str]] = None
+    job_name: str = ""
     iteration: int = 0
 
     def __post_init__(self):
-        self.run_id = f"{uuid.uuid4()}:{datetime.now().strftime('%Y-%m-%dT%H%M')}"
+        self.substituter = Substituter(self.global_substitutions)
```

Comment:
I'm trying to figure out if it makes sense that this class should have a `Substituter`.. At least if the idea is that it should be `Ensemble`, then I would think it shouldn't have something as dynamic as `Substituter`  (or at least it shouldn't be used outside the class).

Reply:
Curious, is the `global_substitutions` necessary after `__post_init__`? Are we adding properties tot he dataclass that is only required during `init`??

Reply:
> I'm trying to figure out if it makes sense that this class should have a Substituter.. At least if the idea is that it should be Ensemble, then I would think it shouldn't have something as dynamic as Substituter (or at least it shouldn't be used outside the class).

I renamed this back to RunContext, because it was not really an Ensemble (and also a bit at odds with the other Ensemble), so I imagine it more as `EnsembleRunInformation`, in that context the `Substituter` makes sense I think, as it then negates a lot of the switch mechanics that were present before. But happy to discuss alternatives.

> Curious, is the global_substitutions necessary after __post_init__? Are we adding properties tot he dataclass that is only required during init??

Dont think so, changed it to `InitVar`

Reply:
Thanks for the changes. We are now no longer doing changes to the substituter post initialisation of the `RunContext` (The `BatchSimulator` added `GEO_ID` previously). I agree that we want some way of generating the paths in `CreateRunPath`, and it makes sense that the `RunContext` would know this. 

If I would be _very_ picky, I would argue that we don't need to leak the `substituter` out of the `run_context`, hence instead of `run_context.substituter.substitute_real_iter`, the `substituter` could be a private variable in `run_context` and one would call a function in `run_context` that uses the `substituter`. Not of critical importance though